source/common/stats/symbol_table_impl.cc

```diff
@@ -317,6 +317,36 @@ void StatNameStorage::free(SymbolTable& table) {
   bytes_.reset();
 }
 
+StatNameStorageSet::~StatNameStorageSet() {
+  // free() must be called before destructing StatNameStorageSet to decrement
+  // references to all symbols.
+  ASSERT(empty());
+}
+
+void StatNameStorageSet::free(SymbolTable& symbol_table) {
+  // We must free() all symbols referenced in the set, otherwise the symbols
+  // will leak when the flat_hash_map superclass is destructed. They cannot
+  // self-destruct without an explicit free() as each individual StatNameStorage
+  // object does not have a reference to the symbol table, which would waste 8
+  // bytes per stat-name. The easiest way to safely free all the contents of the
+  // symbol table set is to use flat_hash_map::extract(), which removes and
+  // returns an element from the set without destructing the element
+  // immediately. This gives us a chance to call free() on each one before they
+  // are destroyed.
+  //
+  // There's a performance risk here, if removing elements via
+  // flat_hash_set::begin() is inefficient to use in a loop like this. One can
+  // imagine a hash-table implementation where the performance of this
+  // usage-model would be poor. However, tests with 100k elements appeared to
+  // run quickly when compiled for optimization, so at present this is not a
+  // performance issue.
+
+  while (!empty()) {
+    auto storage = extract(begin());
```

Comment:
Is there a way to fully-qualify this as a function within flat_hash_map? Just as an aesthetic thing.